src/mon/OSDMonitor.cc

```diff
@@ -78,9 +78,19 @@
 
 #include "json_spirit/json_spirit_reader.h"
 
+#include <boost/algorithm/string/predicate.hpp>
+
 #define dout_subsys ceph_subsys_mon
 #define OSD_PG_CREATING_PREFIX "osd_pg_creating"
 
+namespace {
+
+const uint32_t MAX_POOL_APPLICATIONS = 4;
```

Comment:
As discussed in the CDM, we want to prevent an end-user doing something silly. In reality, it should really be one app per pool since that's the only tested configuration.